### PR TITLE
[Software] Upsert deployment documentation

### PR DESCRIPTION
**Pull request by jwitz**

### CODE REVIEW

File: software/houston-api.md [hunk 1]
```diff
@@ -171,6 +171,146 @@ mutation CreateDeployment {
 
 Here, `<airflow-executor>` can be `LocalExecutor`, `CeleryExecutor`, or `KubernetesExecutor`.
 
+### Create or update a Deployment with configurations
```
Comment (jwitz) [inline software/houston-api.md:174]: Is the `CreateDeployment` mutation deprecated now?

Comment (Simpcyclassy) [reply software/houston-api.md:174]: No it is not.

File: software/houston-api.md [hunk 2]
```diff
@@ -171,6 +171,146 @@ mutation CreateDeployment {
...
+
+The `upsertDeployment` mutation can be used to both create and update Deployments with all possible Deployment configurations. If you query `upsertDeployment` with a `deploymentUuid` that doesn't exist, the Houston API creates a new Deployment according to your specifications. If the `deploymentUuid` does exist, the Houston API updates a Deployment with that ID.
```
Comment (rishkarajgi) [inline software/houston-api.md:176]: `If you query `upsertDeployment` with a `deploymentUuid` that doesn't exist, the Houston API creates a new Deployment according to your specifications`

Correction: 
1. If queried with a deploymentUuid value that doesn't exist then the mutation will throw an error. 
2. If it is blank/empty string it creates a new deployment
3. If deploymentUuid exists then it goes to the update flow

cc: @Priyanka13Astro @Simpcyclassy

Comment (Simpcyclassy) [reply software/houston-api.md]: 1. Correct
2. You need to add a `workspaceUuid` before it creates
3. Correct

Summarily once `deploymentUuid` is passed in the query it goes through the update flow, and if not it goes through the create flow. But for a create flow to be successful you have to pass a `workspaceUuid`.

File: software/houston-api.md [hunk 3]
```diff
@@ -171,6 +171,146 @@ mutation CreateDeployment {
...
+### Create or update a Deployment with configurations
+
+The `upsertDeployment` mutation can be used to both create and update Deployments with all possible Deployment configurations. If you query `upsertDeployment` without a `deploymentUuid`, the Houston API creates a new Deployment according to your specifications. If you specify an existing `deploymentUuid`, the Houston API updates the Deployment with that ID.
+
+The following query creates a new Deployment in a custom namespace `test-new-dep` and configures a Deployment environment variable `AIRFLOW__CORE__COLORED_LOG_FORMAT`.
+
+```graphql
+mutation upsertDeployment(
+  $workspaceUuid: Uuid,
+	$deploymentUuid: Uuid,
```
Comment (Simpcyclassy) [inline software/houston-api.md:183]: Minor fix, check indentation.

File: software/houston-api.md [hunk 4]
```diff
@@ -171,6 +171,146 @@ mutation CreateDeployment {
...
+	"environmentVariables": [
+    {
+      "key": "AIRFLOW__CORE__COLORED_LOG_FORMAT",
+      "value": "test",
+      "isSecret": false
+    }
+  ],
+  "releaseName": "",
+  "namespace": "test-new-dep",
+  "type": "airflow",
```
Comment (Simpcyclassy) [inline software/houston-api.md:286]: I have updated the notion doc, we no longer need to pass `type`

**Comment by Simpcyclassy** [comment]
You can take more details from [here](https://www.notion.so/astronomerio/Upsert-Deployment-91f187aeb9a14b7d8ab948b253833cc6?pvs=4#9db41c27c1a64ca5a753a0c39eea7863) on how the query works

**Comment by Simpcyclassy** [comment]
Based on existing scope, `upsertDeployment` covers
- createDeployment
- updateDeployment
- updateDeploymentVariables
- updateDeploymentImage
- updateDeploymentRuntime


Create flow
- createDeployment
- updateDeploymentVariables

Update Flow
- updateDeployment
- updateDeploymentVariables
- updateDeploymentImage
- updateDeploymentRuntime

File: software/houston-api.md [hunk 5]
```diff
@@ -171,6 +171,146 @@ mutation CreateDeployment {
 
 Here, `<airflow-executor>` can be `LocalExecutor`, `CeleryExecutor`, or `KubernetesExecutor`.
 
+### Create or update a Deployment with configurations
+
+The `upsertDeployment` mutation can be used to both create and update Deployments with all possible Deployment configurations. If you query `upsertDeployment` without a `deploymentUuid`, the Houston API creates a new Deployment according to your specifications. If you specify an existing `deploymentUuid`, the Houston API updates the Deployment with that ID.
```
Comment (jwitz) [inline software/houston-api.md:176]:
```suggestion
The `upsertDeployment` mutation can be used to both create and update Deployments with all possible Deployment configurations. If you query `upsertDeployment` without a `deploymentUuid`, the Houston API creates a new Deployment according to your specifications. If you specify an existing `deploymentUuid`, the Houston API updates the Deployment with that ID. All queries require specifying a `workspaceUuid`.
```

File: software/houston-api.md [hunk 6]
```diff
@@ -171,6 +171,146 @@ mutation CreateDeployment {
...
+	"environmentVariables": [
+    {
+      "key": "AIRFLOW__CORE__COLORED_LOG_FORMAT",
+      "value": "test",
+      "isSecret": false
+    }
+  ],
+  "releaseName": "",
+  "namespace": "test-new-dep",
+  "type": "airflow",
```
Comment (jwitz) [inline software/houston-api.md:286]:
```suggestion
```

File: software/houston-api.md (hunk 5 above)
Comment (Simpcyclassy) [reply software/houston-api.md:176]: Not all queries require `workspaceUuid`, only create path. And when provided along with `deploymentUuid`, `workspaceUuid` is ignored. The existing write-up is okay.

**Pull request merged**